test/server/runner-test.js

```diff
@@ -138,8 +138,73 @@ describe('Runner', function () {
                 .catch(done);
         });
 
+        it('Should raise error if browser disconnected before bootstrapping', function (done) {
+            var brokenConnection = testCafe.createBrowserConnection();
+
+            brokenConnection.establish('Mozilla/5.0 (Macintosh; Intel Mac OS X 10_10_1) AppleWebKit/537.36 ' +
+                                       '(KHTML, like Gecko) Chrome/41.0.2227.1 Safari/537.36');
+
+            brokenConnection.on('error', function () {
+                runner
+                    .run()
+                    .then(function () {
+                        throw new Error('Promise rejection expected');
+                    })
+                    .catch(function (err) {
+                        expect(err.message).eql('The Chrome 41.0.2227 / Mac OS X 10.10.1 browser disconnected. ' +
+                                                'This problem may appear when a browser hangs or is closed, ' +
+                                                'or due to network issues.');
+                    })
+                    .then(done)
+                    .catch(done);
+            });
+
+            runner.
+                browsers(brokenConnection)
+                .reporter('list')
+                .src('test/server/data/test-suite/top.test.js');
+
+            brokenConnection.emit('error', 'It happened');
+        });
+
+        it('Should raise error if multiple browsers disconnected before bootstrapping', function (done) {
```

Comment:
Should raise an error if multiple browsers get disconnected before bootstrapping